components/new_post/post_component.tsx

```diff
@@ -0,0 +1,494 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {MouseEvent, useCallback, useEffect, useRef, useState} from 'react';
```

Comment:
Also, could you merge in the latest master and re-run ESLint? It's going to complain about there not being a space with the change I made